doc/netplan.md

```diff
@@ -340,6 +340,90 @@ These options are available for all types of interfaces.
      :    Match this policy rule based on the type of service number applied to
           the traffic.
 
+## Authentication
+Netplan supports advanced authentication settings for ethernet and wifi
+interfaces, as well as individual wifi networks, by means of the ``auth`` block.
+
+``auth`` (mapping)
+
+:    Specifies authentication settings for a device of type ``ethernets:`` or
+     ``wifis:``, or a single wifi ``access-points:`` entry. If ``auth`` blocks
+     are specified on both a wifi device and one of its access points, only the
+     settings in the access point's ``auth`` block are taken into account;
+     however, the ``auth`` block of the wifi device, if present, is used for any
+     ``access-points:`` without their own ``auth`` block.
+
+     Example:
+
+          wifis:
+            wlp1s0:
+              auth:
+                key-mgmt: wpa-psk
+                psk: "d3f4ul7"
+              access-points:
+                office:
+                  auth:
+                    # the authentication settings above are ignored for the
+                    # wifi named "office"
+                    key-mgmt: wpa-eap
+                    eap-method: ttls
+                    anonymous-identity: "@internal.example.com"
+                    identity: "joe@internal.example.com"
+                    password: "v4ryS3kr1t"
+                home:
+                  mode: infrastructure
+                  # the authentication settings above are applied to the wifi
+                  # named "home"
+
+      The ``auth`` block supports the following properties:
+
+      ``key-mgmt`` (scalar)
+      :    The supported key management modes are ``none`` (no key management),
+           ``wpa-psk`` (WPA with pre-shared key, common for home wifi),
+           ``wpa-eap`` (WPA with EAP, common for enterprise wifi) and ``8021x``
+           (used primarily for wired Ethernet connections).
+
+      The following properties can be used if ``key-mgmt`` is ``wpa-psk``:
+
+      ``psk`` (scalar)
+      :    The pre-shared key for this connection. Useful only if ``key-mgmt``
+           is ``wpa-psk``.
+
+      ``eap-method`` (scalar)
+      :    The supported EAP methods are ``tls`` (TLS), ``peap`` (Protected EAP)
+           and ``ttls`` (Tunneled TLS). Useful only if ``key-mgmt`` is
+           ``wpa-eap`` or ``8021x``.
```

Comment:
This doesn't match the preceding header.  If it's only valid fow wpa-eap and 8021x, then it should be in the section below, and the last sentence can be dropped.

Reply:
Let's name this just "method", please.

Reply:
Oops, that was a relic from the initial structure of the documentation before I decided to insert those headers.

Reply:
@cyphermox how confident can we be that we will never need to support other types of "methods" on 802.1x authentication besides EAP methods (possibly at the same time)?  How clear is it to the user that "method" refers to EAP without the prefix?

Reply:
It's a convenient way to refer to these (commonly called EAP methods) and being more obvious than "authentication"; you could have similar deeper-level config options for other modes aside from EAP -- but right now you only have a choice of using "eap" or "psk".

https://en.wikipedia.org/wiki/Extensible_Authentication_Protocol